src/many-ledger/src/storage.rs

```diff
@@ -171,6 +181,45 @@ impl LedgerStorage {
         self.latest_tid.clone()
     }
 
+    pub fn create_snapshot(&self, height: u64) -> Result<(), ManyError> {
+        let snapshot = SystemTime::now()
+            .duration_since(SystemTime::UNIX_EPOCH)
+            .map_err(|e| ManyError::unexpected_transport_error(e.to_string()))?
+            .as_millis()
+            .to_string();
+
+        let s: String = rand::thread_rng()
+            .sample_iter(&Alphanumeric)
+            .take(7)
+            .map(char::from)
+            .collect();
+        let snapshot_name = format!("{}-{}-{}", "snapshot", snapshot, s);
+
+        self.persistent_store
+            .snapshot(
+                self.snapshot_path
+                    .join(format!("{}-{}", snapshot_name, height)),
+            )
+            .map_err(|e| ManyError::unexpected_transport_error(e.to_string()))?;
+
+        let gz = File::create(
+            self.snapshot_path
+                .join(format!("{}-{}.tar.gz", snapshot_name, &height)),
+        )
+        .map_err(|e| ManyError::unexpected_transport_error(e.to_string()))?;
+
+        let encoder = GzEncoder::new(gz, Compression::default());
+        let mut tar = tar::Builder::new(encoder);
+        tar.append_dir_all(
+            &snapshot_name,
+            self.snapshot_path
+                .join(format!("{}-{}", &snapshot_name, &height)),
+        )
+        .map_err(|e| ManyError::deserialization_error(e.to_string()))?;
+
+        Ok(())
```

Comment:
The last step should probably be to delete the snapshot directory and keep the tar.